File: Emisor/application/application_layer.py
from models.frame import Frame
from application.main_window import MainWindow

class ApplicationLayer:

    def request_message(self):

        window = MainWindow()

        data = window.run()

        frame = Frame()

        frame.message = data["message"]
        frame.algorithm = data["algorithm"]
        frame.destination_port = data["port"]
        frame.ber = data["ber"]

        return frame

File: Receptor/src/application/ApplicationLayer.java
package application;

import models.Frame;

/**
 * Capa de Aplicación del receptor: muestra el mensaje final al usuario,
 * o un mensaje de error si se detectaron errores que no se pudieron corregir.
 * Espejo de Emisor/application/application_layer.py (servicio mostrar_mensaje).
 */
public class ApplicationLayer {

    public void showMessage(Frame frame) {
        System.out.println("\n========== APPLICATION ==========");
        System.out.println("Algoritmo utilizado : " + frame.algorithm);
        System.out.println("Detalle             : " + frame.errorDetail);

        if (frame.corrupted && !frame.corrected) {
            System.out.println("\n*** ERROR: se detectaron errores en la trama y no fue posible corregirlos. ***");
            return;
        }

        if (frame.corrupted && frame.corrected) {
            System.out.println("(Se detectó un error y fue corregido automáticamente.)");
        }

        System.out.println("\nMensaje recibido: " + frame.message);
    }
}
